gson/src/main/java/com/google/gson/internal/bind/CollectionTypeAdapterFactory.java

```diff
@@ -58,6 +64,20 @@ public CollectionTypeAdapterFactory(ConstructorConstructor constructorConstructo
     return result;
   }
 
+  private boolean isRawTypeSupported(Class<?> rawType) {
+    if (supportIterable) {
+      return Iterable.class.isAssignableFrom(rawType);
+    }
+    return Collection.class.isAssignableFrom(rawType);
+  }
+
+  private Type getElementType(Type type, Class<?> rawType) {
+    if (supportIterable) {
+      return $Gson$Types.getIterableElementType(type, rawType);
+    }
+    return $Gson$Types.getCollectionElementType(type, rawType);
```

Comment:
Not sure, but maybe calling `$Gson$Types.getIterableElementType(type, rawType);` would be enough (would also not require the separate method `getElementType` here then), since `Collection<E> extends Iterable<E>`.

Reply:
@Marcono1234 
The `getIterableElementType` and `getCollectionElementType` do slightly different things, so I only dispatch their calls by checking the flag.

Reply:
Yes, their code differs, but I think it should be possible to use `getIterableElementType` insteadof `getCollectionElementType` (given that the `$Gson$Types` methods work as documented).

This would then eliminate the need to register `CollectionTypeAdapterFactory` twice for Gson.